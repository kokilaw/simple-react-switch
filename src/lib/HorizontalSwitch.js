import React, { Component } from "react";
import PropTypes from "prop-types";

import { Blob, BlobChecked } from "./animations";
import { HorizontalWrapper } from "./wrappers";

import hexColorPropType from "./util/hexColorPropType";
import twoParamFuncPropType from "./util/twoParamFuncPropType";

class HorizontalSwitch extends Component {
  render() {
    let { bgUnCheckedColor, bgCheckedColor } = this.props;

    return (
      <HorizontalWrapper
        blob={Blob}
        blobChecked={BlobChecked}
        colorOff={bgUnCheckedColor}
        colorOn={bgCheckedColor}
      >
        <input
          type="checkbox"
          // defaultChecked={defaultChecked}
          // disabled={disabled}
          // onChange={this.handleChange}
        />
        <span />
      </HorizontalWrapper>
    );
  }
}

HorizontalSwitch.propTypes = {
  defaultChecked: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: twoParamFuncPropType,
  key: PropTypes.string,
  bgCheckedColor: hexColorPropType,
  bgUnCheckedColor: hexColorPropType,
  foregroundColor: hexColorPropType
};

HorizontalSwitch.defaultProps = {
  defaultChecked: false,
  disabled: false,
  bgCheckedColor: "#FF4651",
  bgUnCheckedColor: "#48EA8B",
  foregroundColor: "#fff",
  key: "sample-key"
};

export default HorizontalSwitch;
