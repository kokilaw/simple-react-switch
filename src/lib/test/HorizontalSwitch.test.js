import React from "react";
import HorizontalSwitch from "../HorizontalSwitch";
import renderer from "react-test-renderer";

describe("HorizontalSwitch", () => {
  it("renders properly", () => {
    const tree = renderer.create(<HorizontalSwitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
