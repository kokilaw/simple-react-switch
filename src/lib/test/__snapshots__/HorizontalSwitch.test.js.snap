// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalSwitch renders properly 1`] = `
<label
  className="sc-bdVaJa cPwplN"
>
  <input
    type="checkbox"
  />
  <span />
</label>
`;
